tests/NATS.Client.Core.Tests/TlsOptsTest.cs

```diff
@@ -0,0 +1,195 @@
+            KeyFile = clientKeyFile,
+        });
+
+        await ValidateAsync(new NatsTlsOpts
+        {
+            LoadClientCert = NatsTlsOpts.LoadClientCertFromPem(await File.ReadAllTextAsync(clientCertFile), await File.ReadAllTextAsync(clientKeyFile)),
+        });
+
+        await ValidateAsync(new NatsTlsOpts
+        {
+            ConfigureClientAuthentication = async options =>
+            {
+                options.LoadClientCertFromPem(await File.ReadAllTextAsync(clientCertFile), await File.ReadAllTextAsync(clientKeyFile));
+            },
+        });
+
+        return;
+
+        static async Task ValidateAsync(NatsTlsOpts opts)
+        {
+            var clientOpts = await opts.AuthenticateAsClientOptionsAsync(new NatsUri("demo.nats.io", true));
+
+#if NET8_0_OR_GREATER
+            Assert.NotNull(clientOpts.ClientCertificateContext);
+            var leafCert = clientOpts.ClientCertificateContext.TargetCertificate;
+
+            leafCert.Subject.Should().Be("CN=client");
+            var encryptValue = leafCert.GetRSAPublicKey()!.Encrypt(Encoding.UTF8.GetBytes("test123"), RSAEncryptionPadding.OaepSHA1);
+            var decryptValue = leafCert.GetRSAPrivateKey()!.Decrypt(encryptValue, RSAEncryptionPadding.OaepSHA1);
+            Encoding.UTF8.GetString(decryptValue).Should().Be("test123");
+#else
+            Assert.NotNull(clientOpts.ClientCertificates);
+            Assert.Single(clientOpts.ClientCertificates);
+#endif
+        }
+    }
+
+    [Fact]
+    public async Task Load_client_cert_chain_and_key()
+    {
+        const string clientCertFile = "resources/certs/chainedclient-cert.pem";
+        const string clientKeyFile = "resources/certs/chainedclient-key.pem";
+
+        await ValidateAsync(new NatsTlsOpts
+        {
+            CertFile = clientCertFile,
+            KeyFile = clientKeyFile,
+        });
+
+        await ValidateAsync(new NatsTlsOpts
+        {
+            ConfigureClientAuthentication = async options =>
+            {
+                options.LoadClientCertFromPem(await File.ReadAllTextAsync(clientCertFile), await File.ReadAllTextAsync(clientKeyFile));
+            },
+        });
+
+        return;
+
+        static async Task ValidateAsync(NatsTlsOpts opts)
+        {
+#if NET8_0_OR_GREATER
+            var clientOpts = await opts.AuthenticateAsClientOptionsAsync(new NatsUri("demo.nats.io", true));
+            var ctx = clientOpts.ClientCertificateContext;
+            Assert.NotNull(ctx);
+
+            var leafCert = ctx.TargetCertificate;
+            leafCert.Subject.Should().Be("CN=leafclient");
+            var encryptValue = leafCert.GetRSAPublicKey()!.Encrypt(Encoding.UTF8.GetBytes("test123"), RSAEncryptionPadding.OaepSHA1);
+            var decryptValue = leafCert.GetRSAPrivateKey()!.Decrypt(encryptValue, RSAEncryptionPadding.OaepSHA1);
+            Encoding.UTF8.GetString(decryptValue).Should().Be("test123");
+
+            Assert.Equal(2, ctx.IntermediateCertificates.Count);
```

Comment:
On Windows fails here with 3 intermediate certs, last one being CA.

Reply:
It looks like when creating the context from intermediates (I don't know how) the root CA is populated and then removed on Linux but not on Windows.

https://github.com/dotnet/runtime/blob/223e9e71954433544106d5c90e06b5c8113e51d9/src/libraries/System.Net.Security/src/System/Net/Security/SslStreamCertificateContext.cs#L77

https://github.com/dotnet/runtime/blob/223e9e71954433544106d5c90e06b5c8113e51d9/src/libraries/System.Net.Security/src/System/Net/Security/SslStreamCertificateContext.Windows.cs#L12

https://github.com/dotnet/runtime/blob/223e9e71954433544106d5c90e06b5c8113e51d9/src/libraries/System.Net.Security/src/System/Net/Security/SslStreamCertificateContext.Linux.cs#L24

Reply:
it's only happening on my PC.